src/main/java/com/surveyMonkey/controllers/AdminController.java

```diff
@@ -64,13 +62,18 @@ public String showSurvey(@PathVariable String surveyCode, Model model) {
 
     @PostMapping({"/surveyResults"})
     public String surveyResult(@RequestParam("surveyCode") String surveyCode, @RequestParam("surveyPassword") String surveyPassword, Model model) {
-        model.addAttribute("surveyCode", surveyCode);
+        for(Survey survey:surveyRepository.findAll()){
+            if(survey.getSurveyCode().equals(surveyCode)&&survey.getSurveyPassword().equals(surveyPassword)){
+                model.addAttribute("surveyCode", surveyCode);
+                break;
+            }
+        }
         return "results";
     }
 
     @PostMapping({"/surveyResult"})
```

Comment:
Maybe change the name of this postmapping to be more informative or different than the previous one (/surveyResult vs /surveyResults) can be easy to get confused and make mistakes. Not that serious though.